echopop/live/live_data_loading.py

```diff
@@ -0,0 +1,928 @@
+            for file in acoustic_files
+        )
+    ]
+
+    # Concatenate the dataframe component
+    acoustic_data_df = pd.concat([df for df, _ in results_list], ignore_index=True)
+    # ---- Add the `acoustic_data_units` to the dictionary and output the resulting tuple
+    return acoustic_data_df, results_list[0][1] if results_list else None
+
+
+def filter_filenames(
+    directory_path: Path, filename_id: str, files: List[Path], file_extension: str
+):
+
+    # Drop the `{FIELD_ID}` tag identifier
+    file_id_format = re.sub(r"\{FILE_ID:([^}]+)\}", r"\1", filename_id)
+    # ---- Replace all other tags with `*` placeholders
+    file_id_format = re.sub(r"\{[^{}]+\}", "*", file_id_format)
+    # ---- Compile the pattern
+    escaped_file_id_format = re.escape(file_id_format)
+    pattern = re.compile(escaped_file_id_format.replace(r"\*", ".*"))
+    # pattern = re.compile(rf'{file_id_format.replace(".", r"\.").replace("*", ".*")}')
+    # ---- Create Path object with the generalized format: S3
+    s3_files = [
+        filename for filename in files if filename.startswith("s3://") and pattern.search(filename)
+    ]
+    # ---- Local search
+    local_files = Path(directory_path).glob(f"{file_id_format}.{file_extension}")
+    # ---- Assign to subfile path object
+    if s3_files:
+        subfile_path_obj = s3_files
+    else:
+        subfile_path_obj = local_files
+    # ---- List all files that match this pattern
+    subfile_str = [str(file) for file in list(subfile_path_obj)]
+
+    # Convert list of proposed files from Path to String
+    file_str = [str(file) for file in list(files)]
+
+    # Find intersection with the proposed filenames and return the output
+    return list(set(subfile_str).intersection(set(file_str)))
+
+
+def read_biology_files(
+    biology_files: List[str], file_configuration: dict, pandas_kwargs: dict = {}
+):
+
+    # Get the biology data file settings
+    file_settings = file_configuration["input_directories"]["biology"]
+
+    # Get the file-specific settings, datatypes, columns, etc.
+    # ---- Get defined columns and datatypes from `LIVE_INPUT_FILE_CONFIG_MAP`
+    biology_config_map = LIVE_INPUT_FILE_CONFIG_MAP["biology"]
+    # ---- Extract the expected file name ID's
+    biology_file_ids = file_settings["file_name_formats"]
+    # ---- Extract all of the file ids
+    biology_config_ids = list(biology_file_ids.keys())
+    # ---- Initialize the dictionary that will define this key in the `input` attribute
+    biology_output = {f"{key}_df": pd.DataFrame() for key in biology_config_ids}
+    # # ---- Create filepath object
+    if "data_root_dir" in file_configuration:
+        # directory_path = Path(file_configuration["data_root_dir"]) / file_settings["directory"]
+        directory_path = "/".join([file_configuration["data_root_dir"], file_settings["directory"]])
+    else:
+        directory_path = file_settings["directory"]
```

Comment:
I am curious what the problem with using Path object is. Thinking that `Path("") / file_settings["directory"]` would still give Path(file_settings["directory"]) so seems can eliminate the if-else